kyaml/fn/framework/command/example_test.go

```diff
@@ -0,0 +1,383 @@
+	}
+	type ExampleServiceGenerator struct {
+		Spec Spec `yaml:"spec,omitempty"`
+	}
+	functionConfig := &ExampleServiceGenerator{}
+
+	// function implementation -- generate a Service resource
+	p := &framework.SimpleProcessor{
+		Config: functionConfig,
+		Filter: kio.FilterFunc(func(items []*yaml.RNode) ([]*yaml.RNode, error) {
+			var newNodes []*yaml.RNode
+			for i := range items {
+				meta, err := items[i].GetMeta()
+				if err != nil {
+					return nil, err
+				}
+
+				// something we already generated, remove it from the list so we regenerate it
+				if meta.Name == functionConfig.Spec.Name &&
+					meta.Kind == service &&
+					meta.APIVersion == "v1" {
+					continue
+				}
+				newNodes = append(newNodes, items[i])
+			}
+
+			// generate the resource
+			n, err := yaml.Parse(fmt.Sprintf(`apiVersion: v1
+kind: Service
+metadata:
+ name: %s
+`, functionConfig.Spec.Name))
+			if err != nil {
+				return nil, err
+			}
+			newNodes = append(newNodes, n)
+			return newNodes, nil
+		}),
+	}
+	cmd := command.Build(p, true, false)
+
+	// for testing purposes only -- normally read from stdin when Executing
+	cmd.SetIn(bytes.NewBufferString(`
+apiVersion: config.kubernetes.io/v1alpha1
+kind: ResourceList
+# items are provided as nodes
+items:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: foo
+functionConfig:
+  apiVersion: example.com/v1alpha1
+  kind: ExampleServiceGenerator
+  spec:
+    name: bar
+`))
+
+	// run the command
+	if err := cmd.Execute(); err != nil {
+		panic(err)
+	}
+
+	// Output:
+	// apiVersion: config.kubernetes.io/v1alpha1
+	// kind: ResourceList
+	// items:
+	// - apiVersion: apps/v1
+	//   kind: Deployment
+	//   metadata:
+	//     name: foo
+	// - apiVersion: v1
+	//   kind: Service
+	//   metadata:
+	//     name: bar
+	// functionConfig:
+	//   apiVersion: example.com/v1alpha1
+	//   kind: ExampleServiceGenerator
+	//   spec:
+	//     name: bar
+}
+
+// ExampleBuild_generateUpdate generates a resource, updating the previously generated
+// copy rather than replacing it.
+//
+// Note: This will keep manual edits to the previously generated copy.
+func ExampleBuild_generateUpdate() {
+	// function API definition which will be parsed from the ResourceList.FunctionConfig
+	// read from stdin
+	type Spec struct {
+		Name        string            `yaml:"name,omitempty"`
+		Annotations map[string]string `yaml:"annotations,omitempty"`
+	}
+	type ExampleServiceGenerator struct {
+		Spec Spec `yaml:"spec,omitempty"`
+	}
+	functionConfig := &ExampleServiceGenerator{}
+
+	// function implementation -- generate or update a Service resource
+	fn := func(items []*yaml.RNode) ([]*yaml.RNode, error) {
+		var found bool
+		for i := range items {
+			meta, err := items[i].GetMeta()
+			if err != nil {
+				return nil, err
+			}
+
+			// something we already generated, reconcile it to make sure it matches what
+			// is specified by the FunctionConfig
+			if meta.Name == functionConfig.Spec.Name &&
```

Comment:
nit: Reverse this condition so you can `continue` and simplify the loop below.

Reply:
Tried this out, and I think I prefer it as-is, because what this whole loop is doing is looking for this object. Once we find it, we don't need to check any more items, so we still need the `break` anyway.